expression/builtin_string.go

```diff
@@ -1002,39 +1003,57 @@ type hexFunctionClass struct {
 }
 
 func (c *hexFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinHexSig{newBaseBuiltinFunc(args, ctx)}
-	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
+	switch t := args[0].GetTypeClass(); t {
+	case types.ClassString:
+		bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		// Use UTF-8 as default
```

Comment:
hex(str) will return the hex string of the original string byte by byte, so for utf-8 string, it should be *3 for each rune and *2 for each byte.